Add UPE opt-in banner in stripe settings page

Mayisha opened the pull request:
Fixes #1715

### Description
Showing the UPE opt-in banner in the main Stripe settings page. The payment method section is not implemented yet. A placeholder with card information has been added there. The main implementation of the payment methods will be done in #1716

### Changes in this PR
- Reused the `UPEOptInBanner` component 
- Added a placeholder over the banner with card information. 
- Added tests

![banner in stripe settings](https://user-images.githubusercontent.com/33387139/131354937-7cff38f1-038c-4f15-87e6-2f913a4e2e0b.png)

### Testing instructions
- Set `_wcstripe_feature_upe_settings` flag to `1` and go to the `Stripe` settings page. The banner should appear beside the payment accepted on the checkout section.

### QA steps

- [ ] The old Stripe settings page does not have the banner
- [ ] Set `_wcstripe_feature_upe_settings` flag to `1`. (Returning `true` from [this function](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/24277919aa7b9029c39ecd997dd133990e7f65c7/includes/class-wc-stripe-feature-flags.php#L34) should do the work). The new Stripe settings page has the banner
- [ ] All the tests are passing

Mayisha committed:
Add UPE opt-in banner in the Stripe settings page

Mayisha committed:
Add tests for UPE opt-in banner in the Stripe settings page

waclawjacek commented on client/settings/general-settings-section/index.js:
```diff
@@ -2,52 +2,85 @@
+					<CardsIcon size="medium" />
+					<PaymentMethodText>
+						<PaymentMethodLabel>
+							Credit card / debit card
+						</PaymentMethodLabel>
+						<PaymentMethodDescription>
+							Let your customers pay with major credit and debit
+							cards without leaving your store.
```
Make sure to i18n the strings:

```js
import { __ } from '@wordpress/i18n';

{ __( 'Credit card / debit card', 'woocommerce-gateway-stripe' ) }
```

waclawjacek commented on client/settings/general-settings-section/index.js:
```diff
@@ -2,52 +2,85 @@
+					</PaymentMethodText>
+				</CardBodyWrapper>
+			</Card>
+			<OptInBannerWrapper data-testid="opt-in-banner">
```
Let's add "UPE" to the name to make it a bit clearer what banner this is:

```suggestion
			<UPEOptInBannerWrapper data-testid="upe-opt-in-banner">
```

waclawjacek commented on client/settings/general-settings-section/__tests__/general-settings-section.test.js:
```diff
@@ -0,0 +1,35 @@
+
+		expect( screen.queryByTestId( 'opt-in-banner' ) ).toBeInTheDocument();
+		expect(
+			screen.queryByText( 'Enable in your store' )
```
This is 100% OK but for a little extra benefit you could also check if what is rendered is a button:

```suggestion
			screen.queryByRole( 'button', { name: 'Enable in your store' } )
```

Mayisha replied:
This element is found by role `link`. Updated the test with this-
`screen.queryByRole( 'link', { name: 'Enable in your store' } )`

waclawjacek reviewed (changes requested):
Great PR! :tada: I really like the code cleanliness and how everything is tested. I posted some minor comments.

It seems some of the functionality described in the issue's "Acceptance criteria" section is still missing from the PR, namely:

* Banner redirects to the onboarding task
  * Looks like the URL is `http://localhost:8082/wp-admin/admin.php?page=wc_stripe-onboarding_wizard`
* Banner is not displayed when UPE is enabled

Could you please add these things?

Mayisha committed:
Address Feedbacks.
- Update store link
- Update wrapper name to 'UPEOptInBannerWrapper'
- Assert button by role in test
- Hide banner if UPE is enabled. Use 'isUpeEnabled' field from 'UpeToggleContext' for this
- Add additional test for banner

Mayisha commented:
@waclawjacek thank you for the feedback. Please review [this commit](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1764/commits/e1ad3edfa3025622fdd6ab43d2bf497d4fd34de5) where I have addressed the feedback and let me know if it works :panda_face:

waclawjacek reviewed (approved):
LGTM! Thank you for including the changes! :shipit:

Mayisha committed:
Merge branch 'develop' into add/1715-upe-opt-in-banner-in-stripe-settings
